### ISSUE TITLE
most URLs sent to Wayback have 502 Bad Gateway status

### ISSUE THREAD

**Issue by milliken** [buren/wayback_archiver#10]
Running Wayback_archiver on https://www.bia.gov (and also on one of my own sites as a test) I see that most of the URLS in the list sent to the internet archive have 502 Bad Gateway listed in the status.  A few give 200s and a few 404s too so it isn't everything.  Is this expected or a problem with the site itself?

Found: https://www.bia.gov/training/AOTR/index.html
Found: https://www.bia.gov/training/PL638/index.html
=== WAYBACK ARCHIVER ===
Request are sent with up to 10 parallel threads
Total urls to be sent: 785
[502, Bad Gateway] https://www.bia.gov/WhoWeAre/index.htm
[502, Bad Gateway] https://www.bia.gov/News/index.htm
[502, Bad Gateway] https://www.bia.gov/WhereIsMy/index.htm
[502, Bad Gateway] https://www.bia.gov/Calevents/index.htm

[502, Bad Gateway] https://www.bia.gov/WhoWeAre/BIA/index.htm
[502, Bad Gateway] https://www.bia.gov/WhoWeAre/AS-IA/OHSEM/index.htm
[200, OK] https://www.bia.gov/WhoWeAre/AS-IA/CLA/index.htm
[502, Bad Gateway] https://www.bia.gov/WhoWeAre/AS-IA/OHCM/index.htm

**Comment by buren** [comment]
Hm.. interesting I actually test crawled `bls.gov` and didn't see any `502`s. I'll have to dig deeper.

Thanks for reporting!

**Comment by buren** [comment]
@milliken after some digging it seems like its actually Internet Archive that returns that status and its not a problem with this gem.. 

This URL works fine: https://www.bia.gov/WhatWeDo/index.htm, however the URL at the Internet Archive:
https://web.archive.org/save/https://www.bia.gov/WhatWeDo/index.htm returns an error. 

Here is another example 
- https://www.bia.gov/WhatWeDo/ServiceOverview/InformationTechnology/index.htm
- https://web.archive.org/save/https://www.bia.gov/WhatWeDo/ServiceOverview/InformationTechnology/index.htm

## Screenshot (Internet Archive)

<img width="533" alt="screen shot 2017-02-20 at 23 33 39" src="https://cloud.githubusercontent.com/assets/922411/23144022/085f5110-f7c5-11e6-8fc6-60fd8aed17ae.png">

---

I've released a new version, `v0.1` of `wayback_archiver`, that posts the URL directly when found. The current version tries to find all URLs first, *which may never finish* ⚠️ , and then post them in parallel to the Internet Archive.

In other words I ~strongly~ kindly suggest you upgrade to the new version `gem install wayback_archiver`.

Thanks again for reporting bugs! 👍

**Comment by buren** [comment]
@milliken I've now tested `WaybackArchiver` against bia.gov again and all URLs are posted successfully 🎉 

* I've just added a major performance improvement by implementing a thread pool that process the found URLs and posts them to the Wayback Machine, see #12.

* I've also implemented better and configurable logging, see #13.

*  I've exposed another top-level method `WaybackArchiver#crawl` where you can configure the concurrency, see [README.md#usage](https://github.com/buren/wayback_archiver/#usage).
    ```ruby
    WaybackArchiver.crawl(site, concurrency: 7) # default is 5
    ```

Finally I've released a new version 0.2.0 of the gem to RubyGems.
